crates/router/src/core/payments/helpers.rs

```diff
@@ -610,10 +610,37 @@ pub async fn get_customer_from_details(
 pub async fn get_connector_default(
     merchant_account: &storage::MerchantAccount,
     state: &AppState,
+    payment_method: Option<api::PaymentMethod>,
     request_connector: Option<api_enums::Connector>,
 ) -> CustomResult<api::ConnectorCallType, errors::ApiErrorResponse> {
     let connectors = &state.conf.connectors;
-    if let Some(connector) = request_connector {
+    if matches!(
+        payment_method,
+        Some(api::PaymentMethod::Wallet(api::WalletData {
+            issuer_name: api_enums::WalletIssuer::Paypal,
+            token: Some(_)
+        }))
+    ) {
+        let connector_data = api::ConnectorData::get_connector_by_name(
+            connectors,
+            "braintree",
+            api::GetToken::Connector,
+        )?;
+        Ok(api::ConnectorCallType::Single(connector_data))
```

Comment:
Can this block just return the connector as &str and then construction of connector be done outside of this block? Code seems to be repeated here